### PR TITLE
Adding pyem to conda-forge

### PR DESCRIPTION
**Pull request by asarnow**
@conda-forge/help-python Hello! I believe my recipe is ready, hopefully I have understood the documentation. The package contains a library and utility functions for image processing in biological electron microscopy, mostly focused on interoperability. It's been cited in a [number of academic publications](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C48&q=10.5281%2Fzenodo.3576630&btnG=).

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by asarnow** [comment]
@conda-forge/help-python I'm fairly confident now, all the automated checks have passed. Thank you for your time!

**Comment by jaimergp** [comment]
The logs show no builds 🤔

**Comment by asarnow** [comment]
I misunderstood the selector documentation, apologies.

**Comment by asarnow** [comment]
@jaimergp OK *now* I think I'm good! One of my dependencies (pyfftw on conda-forge) wasn't installing against Python 3.12 so I limited it to 3.9 - 3.11 for now. The same package necessitates skipping Windows.

**Comment by asarnow** [comment]
@conda-forge/help-python ready for review, thanks!

**Comment by asarnow** [comment]
@conda-forge/staged-recipes @conda-forge/help-python
I'm ready for review, apologies for sending another message but I may have fallen off the radar, I can see from the PRs the review team is very busy. Thank you!

**Comment by asarnow** [comment]
@jaimergp sorry to bother you again, just hoping to get a review. I suppose I'll close this and try a new PR soon since it looks like I've been lost in the churn. (And much newer PRs are being accepted).

File: recipes/pyem/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    - healpy >=1.11
+    - natsort >=6.0
+    - starfile >=0.5.2
+
+test:
+  imports:
+    - pyem
+  requires:
+  commands:
+    - csparc2star.py --help
```
Comment (jaimergp) [inline recipes/pyem/meta.yaml:58]: You will want to test all the entry points mentioned above

File: recipes/pyem/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    - pathos >=0.2.1
+    - pyfftw >=0.12
+    - healpy >=1.11
+    - natsort >=6.0
+    - starfile >=0.5.2
+
+test:
+  imports:
+    - pyem
+  requires:
```
Comment (jaimergp) [inline recipes/pyem/meta.yaml:56]:
```suggestion
```

File: recipes/pyem/LICENSE
Comment (jaimergp) [inline recipes/pyem/LICENSE:1]: Delete this file. It's already in the repo.

File: recipes/pyem/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: true  # [win or py<39 or py>311]
+  entry_points:
+    - cfsc.py = pyem.cli.cfsc:_main_
+    - csparc2star.py = pyem.cli.csparc2star:_main_
+    - ctf2star.py = pyem.cli.ctf2star:_main_
+    - disparticle.py = pyem.cli.disparticle:_main_
+    - emcalc.py = pyem.cli.emcalc:_main_
+    - map.py = pyem.cli.map:_main_
```
Comment (jaimergp) [inline recipes/pyem/meta.yaml:22]: Some of these entry points are a bit too generic and cause conflicts with other tools (which also made the same poor choice). Usually a project specific prefix is desired; e.g. `pyem-map`. But I'll leave that to upstream.

File: recipes/pyem/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/asarnow/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: a3e97c2ca83229617b04babf8912ef11e9258b225b2c3e559d654886bdc32d16
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: true  # [win or py<39 or py>311]
```
Comment (jaimergp) [inline recipes/pyem/meta.yaml:15]:
```suggestion
  noarch: python
```

Why only Python 3.10 and Unix? We can express that with requirements metadata (see below), but I want to make sure I understand why this is the case first.

File: recipes/pyem/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    - project.py = pyem.cli.project:_main_
+    - projection_subtraction.py = pyem.cli.projection_subtraction:_main_
+    - stack.py = pyem.cli.stack:_main_
+    - star.py = pyem.cli.star:_main_
+    - star2bild.py = pyem.cli.star2bild:_main_
+    - subparticles.py = pyem.cli.subparticles:_main_
+    - varmap.py = pyem.cli.varmap:_main_
+
+requirements:
+  build:
```
Comment (jaimergp) [inline recipes/pyem/meta.yaml:35]:
```suggestion
```

File: recipes/pyem/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  host:
+    - python
```
Comment (jaimergp) [inline recipes/pyem/meta.yaml:37]:
```suggestion
    - python 3.10.*
```

If you really only want Python 3.10, but I doubt that's the case.

File: recipes/pyem/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    - pip
+  run:
+    - python
```
Comment (jaimergp) [inline recipes/pyem/meta.yaml:40]:
```suggestion
    - python >= 3.8
    - __unix  # avoid Windows installs
```

**Comment by jaimergp** [comment]
Ah ok, this is the reason:

> One of my dependencies (pyfftw on conda-forge) wasn't installing against Python 3.12 so I limited it to 3.9 - 3.11 for now. The same package necessitates skipping Windows.

We usually let this fail because in the future the package might become available and then your package will work. So let's switch to `noarch: python` and implement an open-ended python requirement.

File: recipes/pyem/meta.yaml (hunk 6 above)
Comment (asarnow) [reply recipes/pyem/meta.yaml]: Done

**Comment by asarnow** [comment]
@jaimergp Thank you! I've made all the recommended changes, except for the entry point names (I am upstream). If it's not a blocker I would leave them for now. The users are all biologists, many of them rely on scripts from potentially departed lab members for reproducibility. If that's alright, I can warn my users and change them (plus provide a script to symlink w/ the old names) in a few months.

**Comment by jaimergp** [comment]
> The users are all biologists, many of them rely on scripts from potentially departed lab members for reproducibility

This hits close home hah 🫂

**Pull request merged**